### ISSUE TITLE
Only one time step?

### ISSUE THREAD

**Issue by zsh965866221** [titu1994/LSTM-FCN#4]
There is no doubt that the performance of your models is very good. But I have a little question.

---

The input shape of RNN in keras is that 
> 3D tensor with shape `(batch_size, timesteps, input_dim)`.
[keras RNN](https://keras.io/layers/recurrent/)

The follow is your codes
```python
ip = Input(shape=(1, MAX_SEQUENCE_LENGTH))
x = LSTM(8)(ip)
x = Dropout(0.8)(x)
```
Is there only one timestep in your model, but "sequence length" input dim. So whether this can be directly considered as a fully connected layer represented by a one-timestep `LSTM` layer?

**Comment by titu1994** [comment]
This is referenced in the readme. The paper mentions the dim shuffle being operated on the Input of shape (N, Timesteps, Input dim) and this being passed to the lstm. 

Our original code was written in the inverse way. The input is pre-transposed and of the shape (N, 1, Timesteps) and this is then dim shufed and passed to the CNN instead. 

The two operations are equivalent. Sorry for the confusion. The 1 is from the univariate series, meaning T timesteps and 1 Input dim.

**Comment by FelixHuiweiLin** [comment]
Excuse me, sir. There is no doubt that the performance of your models is very good. But I have the same question as  @zsh965866221 .

As we all know, LSTM is usually used to extract the features of time and the input shape of LSTM in Keras is that:

    3D tensor with shape (batch_size, timesteps, input_dim).

However, you set the timestep as one in your paper and codes such as:

ip = Input(shape=(1, MAX_SEQUENCE_LENGTH))
x = LSTM(8)(ip)
x = Dropout(0.8)(x) 
which means the shape of LSTM's input is (batch_size, 1 , MAX_SEQUENCE_LENGTH)

Emm,  I just want to learn from you that whether it can extract the the features of time if it just has one timestep?

**Comment by titu1994** [comment]
On it's own, this dimension shuffled LSTM is not going to learn features well, which are discriminative enough for a classifier in a downstream task. It is only when we combine it with a CNN feature extraction module that it learns features that are discriminative. 

It cannot however be considered as a simple two later Fully connected subnetwork, since the LSTM gating is different from simple matmul in a Fully connected layer.

**Comment by moorejee** [comment]
HI,
I have a small question. In LSTM model, you set the dropout rate 0.8, is it too large for LSTM to fit the data? I always set the rate 0.1~0.2. So, I would like to know the reason about the choice of it. Thanks a lot.
